js/classes/entities/foreground/weather_system.js

```diff
@@ -0,0 +1,182 @@
+        // Make sure to only have a number of clouds that is divisble by 100. For example, 15 is bad value but 25 is a good value.
+        const numOfClouds = [2, 5, 10, 20, 25, 50];
+        let aboveGroundLevel = 19;
+        // TODO add in dark clouds.
+        if (this.makeRain) {
+            // Add dark clouds
+        } else {
+            // Add white clouds
+        }
+
+        let cloudNum = numOfClouds[this.intensity];
+        // Make so many clouds of each type.
+        for (let cloudType = 0; cloudType < 3; cloudType++) {
+            // Choose a multiplier that spaces out the clouds across the whole zone nicely.
+            let spaceBetweenMultiplier = 0;
+            for (spaceBetweenMultiplier = 0; spaceBetweenMultiplier < 100; spaceBetweenMultiplier++) {
+                if (spaceBetweenMultiplier * cloudNum == 100) {
+                    break;
+                }
+            }
+
+            for (let i = 0; i < cloudNum; i++) {
+                let foreground = 0;
+                // Give the clouds some random variation.
+                let yVariation = Math.random() * (10 - 16) + 14;
+                let xVariation = Math.random() * (20 - 50) + 20;
+                // Choose a location for each cloud that is added such that it has a random x variation and it is spaced out over the zone.
+                // i * spaceBetweenMultiplier ensure that there will be clouds across the whole zone.
+                let randomOrigin = xVariation + i * spaceBetweenMultiplier;
+                let cloudPosition = Vector.blockToWorldSpace(new Vector(randomOrigin, aboveGroundLevel - yVariation));
+                let cloudVarient = Decoration.DECORATIONS.clouds.CLOUD_BUSHY;
+                switch (cloudType) {
+                    case 1:
+                        cloudVarient = Decoration.DECORATIONS.clouds.CLOUD_LANKY;
+                        break;
+                    case 2:
+                        cloudVarient = Decoration.DECORATIONS.clouds.CLOUD_JUST_CLOUD;
+                        break;
+                    default:
+                        break;
+                }
+                GAME.addEntity(new Decoration(cloudVarient, cloudPosition), foreground);
+
+            }
+        }
+
+    }
+
+    /**
+     * Adds rain or snow.
+     */
+    makePrecipitation() {
+        let dir = ["left", "down", "right", "down", "down"];
+        let rainintensity = [2, 5, 10, 20, 25, 30];
+        let dirIndex = Math.floor(Math.random() * dir.length);
+        let scale = 1;
+
+        for (let j = 0; j < rainintensity[this.intensity]; j++) {
+            for (let i = 0; i < 20; i++) {
+                for (let k = 0; k < 4; k++) {
+                    if (this.type === "snow") {
+                        scale = Math.random() * (.5 - .2) + .2;
+
+                    } else if (dir[dirIndex] !== "down") {
+                        scale = Math.random() * (.8 - .4) + .4;
+                    } else {
+                        r
```

Comment:
What is this?

Reply:
That appears to be a typo, I'm surprised that made it in this PR without causing any issues.